crates/tests/src/integration/masp.rs

```diff
@@ -9365,3 +9365,180 @@ fn multiple_inputs_from_single_note() -> Result<()> {
 
     Ok(())
 }
+
+// Test that shielding and unshielding transactions can pay a small fee to a
+// transparent address as a form of sustainability fee for frontend providers
+#[test]
+fn frontend_sus_fee() -> Result<()> {
```

Comment:
I don't think the code is adequately checking that all of the debits added are less than the user's balance. That check should be implemented by the client and tested here as well.

Reply:
Done in 56418fe8dc0dd1668fcd9c17b31216b7ff7e65a1